src/test/java/io/dropwizard/kafka/integration/AdminClientIntegrationConfiguration.java

```diff
@@ -0,0 +1,34 @@
+package io.dropwizard.kafka.integration;
+
+import java.util.Set;
+
+import javax.validation.constraints.NotNull;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import io.dropwizard.Configuration;
+import io.dropwizard.kafka.BasicKafkaAdminClientFactory;
+import io.dropwizard.kafka.KafkaTopicFactory;
+
+public class AdminClientIntegrationConfiguration extends Configuration {
+
+    @NotNull
+    @JsonProperty("admin")
+    private BasicKafkaAdminClientFactory adminClientFactory;
+
+    @NotNull
+    @JsonProperty("topic")
+    private KafkaTopicFactory topic;
```

Comment:
2 comments:

1. I think the `KafkaTopicFactory` should maybe be contained **in the** `BasicKafkaAdminClientFactory` as a member. 
2. Either the `KafkaTopicFactory` should be able to be `null`, or it should have a boolean flag `enabled` inside of it to allow opting out of that functionality.